Speech Recognition API updates

RaduDumitriu-Dev opened the pull request:
- removed strict aws configuration
- added "unified" api versioning
- added samplerate and encoding into speechrecog Payload
- was not able to add timeInSeconds and confidence float fields. these would cause a crash as both mandatory and optional params

RaduDumitriu-Dev committed:
speech recognition API updates

- removed strict aws configuration
- added "unified" api versioning
- added samplerate and encoding into speechrecog Payload
- was not able to add timeInSeconds and confidence float fields. these should cause a crash as both mandatory and optional params

bensalilijames commented on Source/CharismaModule/Public/CharismaEvents.h:
```diff
@@ -388,79 +388,31 @@ struct FCharismaMessageHistoryResponse
 
-	UPROPERTY(BlueprintReadOnly, Category = "Charisma|Event")
-	FString Transcript;
+	MSGPACK_DEFINE_MAP(message, errorOccuredWhen);
```
to be consistent with other events returned from the server, let's use the convention of mapping lower-case fields from the MsgPack result into upper-case fields in the `struct` (like above)

bensalilijames reviewed (changes requested):
one minor comment, looks great 👍

RaduDumitriu-Dev committed:
Update CharismaEvents.h

- updated CharismaEvent FSpeechRecognitionErrorResult to respect naming convention of server obtained structs
